replication-api-impl/src/main/java/com/connexta/ion/replication/api/impl/data/SiteQueryIndexImpl.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.ion.replication.api.impl.data;
+
+import com.connexta.ion.replication.api.data.SiteQueryIndex;
+import java.time.Instant;
+import java.util.UUID;
+import org.apache.commons.lang3.Validate;
+import org.springframework.data.solr.core.mapping.Indexed;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+/** Simple implementation of SiteQueryIndex. */
+@SolrDocument(collection = "site_query_index")
+public class SiteQueryIndexImpl extends AbstractPersistable implements SiteQueryIndex {
+
+  private static final int CURRENT_VERSION = 1;
+
+  @Indexed(name = "modified_since")
+  private final Instant modifiedSince;
+
+  @Indexed(name = "site_id")
+  private final String siteId;
```

Comment:
Should we be indexing by filter, rather than by site?

Reply:
☝️